include/xtensor/xarray.hpp

```diff
@@ -509,15 +509,12 @@ namespace xt
     }
 
     template <class EC, layout_type L, class SC>
-    inline void xarray_adaptor<EC, L, SC>::assign_temporary_impl(temporary_type& tmp)
+    inline void xarray_adaptor<EC, L, SC>::assign_temporary_impl(temporary_type&& tmp)
     {
-        // TODO (performance improvement) : consider moving tmps
-        // shape and strides
-        base_type::shape_impl() = tmp.shape();
-        base_type::strides_impl() = tmp.strides();
-        base_type::backstrides_impl() = tmp.backstrides();
-        m_data.resize(tmp.size());
-        std::copy(tmp.data().cbegin(), tmp.data().cend(), m_data.begin());
+        base_type::shape_impl() = std::move(tmp.shape());
```

Comment:
I think you should `const_cast` tmp.shape() before moving it, else it will call the copy constructor